src/ray/core_worker/fiber.h

```diff
@@ -158,6 +159,14 @@ class FiberState {
   }
 
  private:
+  // The fiber stack size. 256 KiB.
+  // Note that the default fiber stack size is 128 KiB. In the CI environment, this is not
+  // large enough and thus it causes sefaults. See
+  // https://github.com/ray-project/ray/pull/44292.
+  static constexpr size_t kStackSize = 1024 * 256;
```

Comment:
Actually what's the implication of it? We will use more memory for stack?

Maybe let's just do that for mac since Linux is fine. We have something similar in `core_worker.cc`:

```
#if defined(__APPLE__)
  // io thread will run python code through cython
  // but Mac's default stack size for non-main-thread is too small
  // for certain python libraries like numpy and will cause sigbus.
  // Here we increase the stack size to the size that python uses in
  // https://github.com/python/cpython/blob/v3.9.0/Python/thread_pthread.h#L35.
  // See https://github.com/ray-project/ray/issues/41094 for more details.
  io_thread_attrs.set_stack_size(16777216);
#endif
```

Reply:
We discussed offline the possibility of using a segmented_stack, but it looks like that is not supported on Windows:
```
#if defined(BOOST_USE_SEGMENTED_STACKS)                                                       
# if ! defined(BOOST_WINDOWS)                                                                 
using segmented_stack = boost::context::segmented_stack;                                      
using default_stack = boost::context::default_stack;                                          
# endif                                                                                       
#endif 
```

I would be in favor of keeping the stack size at 256 KiB across all platforms for a couple reasons:
1. If we are exceeding the stack size of Mac, it is likely the other platforms are close to exceeding it as well.
2. It seems cleaner to avoid a platform-specific dependency whenever possible.

Would it be alright to submit this as is and see if it causes any regressions? What do you think?

Reply:
I'm in favor of increasing to 256KiB.